drivers/gpio/gpio_xmc4xxx.c

```diff
@@ -173,6 +211,9 @@ static const struct gpio_driver_api gpio_xmc4xxx_driver_api = {
 	.port_clear_bits_raw = gpio_xmc4xxx_clear_bits_raw,
 	.port_toggle_bits = gpio_xmc4xxx_toggle_bits,
 	.pin_interrupt_configure = gpio_xmc4xxx_pin_interrupt_configure,
+#if defined(CONFIG_XMC4XXX_INTC)
```

Comment:
Here, you could use `IS_ENABLED(CONFIG_XMC4XXX_INTC) ? manage_callback : NULL` which would require fewer `#ifdef`s

Reply:
thanks, I've updated the PR.